config/crd/bases/config.nri_balloonspolicies.yaml

```diff
@@ -195,6 +195,9 @@ spec:
                       items:
                         type: string
                       type: array
+                    preferECores:
+                      description: 'preferECores: prefer efficient cores (E-cores)'
+                      type: boolean
```

Comment:
Couldn't handle this at the user-visible interfaces without explicit reference to P- and E-cores which IMO is a bit HW-/platform-specific. One potential way to do it would be to (re)use instead the CPU priorities `high` and `low`, which are already end-user visible through configuration (used by the existing `AllocatorPriority` option). BTW, this is how a similar preference configuration and corresponding annotations have been added to the topology-aware policy. `high` will prefer P-cores if available, and (together with `normal`) avoid E-cores if possible, while `low` will prefer E-cores if available to fulfill the allocation. This would also has the additional benefit that if you are not running on a hybrid-core system but instead on a system where there is a single kind of core but different cores are clocked differently, allocation will still try to do a reasonable job to a similar effect within the realm of possibilities and considering the different clocking (classes) of the cores.

Reply:
Do you suggest to use AllocatorPriority with values Low, High ?

Reply:
Well, I'm really asking whether we could somehow avoid leaking arguably hardware-specific terminology into the user-visible bits of the interface to the balloons policy, in this case through the configuration ?

So not necessarily directly AllocatorPriority with values low/high, but at least effectively the same approach codewise and similar terminology in the user-visible configuration. But I think using AllocatorPriority with low/high should have the same preference effect as the separate preferences you try to introduce here...

Reply:
I'm fine to use `AllocatorPriority` and perhaps some alternatives I can think of would be `PerformanceMode`, `ExecutionProfile`, `ResourcePriority`, `AllocationWeight`...

Reply:
> similar preference configuration and corresponding annotations have been added to the topology-aware policy

Consistent terminology between CPU selecting functionality would be a big plus, even/especially between different plugins. It avoids human error with less to remember when configuring.

Reply:
@klihub perhaps we use `CoreOptimization` or `CPUUtilizationType` or `CoreUsageMode`.

Reply:
One think to keep in mind though, when having a single key instead of two separate as it is currently on this PR, there won't be an option to opt in for both efficient and performance cores nor will there be a automatic prioritization of performance cores over efficient cores as it is the case right now.

Reply:
But since it is now a string with no (other) default (than the zero-value empty string), you can implement prioritizing performance cores by default, if you think that is the right thing to do. 

About preferring both... that's IMO a bit vague unless you ask for exactly 2 CPUs. Otherwise if you ask for, let's say, 10 CPUs and prefer both what does that mean in practice: 1 power-efficient and 9 performance cores, 9 power-efficient and 1 performance core, any other of the possible combinations ?